packages/main/src/DateRangePicker.js

```diff
@@ -0,0 +1,436 @@
+
+		for (let i = 0; i < dayItems.length; i++) {
+			if ((dayItems[i].dataset.sapTimestamp < firstDateTimestamp && dayItems[i].dataset.sapTimestamp > lastDateTimestamp)
+				|| (dayItems[i].dataset.sapTimestamp > firstDateTimestamp && dayItems[i].dataset.sapTimestamp < lastDateTimestamp)) {
+				dayItems[i].setAttribute("hovered", "");
+			} else {
+				dayItems[i].removeAttribute("hovered");
+			}
+		}
+	}
+
+	_splitValueByDelimiter(value) {
+		let returnValue = [];
+
+		if (!value) {
+			return ["", ""];
+		}
+
+		if (this.delimiter) {
+			returnValue = String(value).split(this.delimiter);
+		}
+
+		return returnValue;
+	}
+
+	setValue(value) {
+		const emptyValue = value === "",
+			isValid = emptyValue || this.isValid(value),
+			isInValidRange = this.isInValidRange(value);
+		let dates = [undefined, undefined];
+
+		if (value === this.value) {
+			return this;
+		}
+
+		if (value) {
+			dates = this._splitValueByDelimiter(value);
+			if (isValid && isInValidRange) {
+				this.valueState("Error");
+				console.warn("Value can not be converted to a valid dates", this); // eslint-disable-line
+			}
+		}
+
+		this._firstDateTimestamp = this._getTimeStampFromString(dates[0]);
+		this._lastDateTimestamp = this._getTimeStampFromString(dates[1]);
+		this._calendar.selectedDates = this.dateIntervalArrayBuilder(this._firstDateTimestamp, this._lastDateTimestamp);
+		this.value = value;
+		this._getInput().setValue(value);
+
+		return this;
+	}
+
+	_changeCalendarSelection(focusTimestamp) {
+		if (this._calendarDate.getYear() < 1) {
+			// 0 is a valid year, but we cannot display it
+			return;
+		}
+
+		const oCalDate = this._calendarDate,
+			timestamp = focusTimestamp || oCalDate.valueOf() / 1000,
+			dates = this._splitValueByDelimiter(this.value);
+
+		this._calendar = Object.assign({}, this._calendar);
+		this._calendar.timestamp = timestamp;
+		if (this.value) {
+			this._calendar.selectedDates = this.dateIntervalArrayBuilder(this._getTimeStampFromString(dates[0]), this._getTimeStampFromString(dates[1]));
+		}
+	}
+
+	get _calendarDate() {
+		const dates = this._splitValueByDelimiter(this.value),
+			millisecondsUTCFirstDate = dates[0] ? this.getFormat().parse(dates[0], true).getTime() : this.getFormat().parse(this.validValue, true).getTime(),
+			oCalDateFirst = CalendarDate.fromTimestamp(
+				millisecondsUTCFirstDate - (millisecondsUTCFirstDate % (24 * 60 * 60 * 1000)),
+				this._primaryCalendarType
+			);
+
+		return oCalDateFirst;
+	}
+
+	/**
+	 * Currently selected first date represented as JavaScript Date instance.
+	 *
+	 * @readonly
+	 * @type { Date }
+	 * @public
+	 */
+	get firstDateValue() {
+		return this.getFormat().parse(this.value);
+	}
+
+	/**
+	 * Currently selected last date represented as JavaScript Date instance.
+	 *
+	 * @readonly
+	 * @type { Date }
+	 * @public
+	 */
+	get lastDateValue() {
+		return this.getFormat().parse(this.value);
+	}
+
+	_handleInputChange() {
+		const nextValue = this._getInput().getInputValue(),
```

Comment:
Is this method _handleInputChange  different from the on in the DatePicker

Reply:
Here the difference is on line 286, but the difference in all of them is the same, so there is an option to override the function i am using  and skip these 3. But a change will occur in other methods.